File: 2021/C#/Day02/Program.cs

using System;
using System.IO;

class Program {

    static void Main(string[] args) {

        List<string> data = BuglLib.Files.ReadFile(args[0]);

        Solve1(data);

        Solve2(data);

    }

    static void Solve1(List<string> data) {

        int pos = 0;
        int depth = 0;

        foreach (string s in data) {
            
            string[] dirX = s.Split(" ");

            switch (dirX[0]) {
                case "forward":
                    pos += int.Parse(dirX[1]);
                    break;
                case "down":
                    depth += int.Parse(dirX[1]);
                    break;
                case "up":
                    depth -= int.Parse(dirX[1]);
                    break;
            }

        }

        Console.WriteLine(String.Format("Part One | Position: {0} | Depth: {1} | Result: {2}", pos, depth, pos * depth));

    }

    static void Solve2(List<string> data) {

        int pos = 0;
        int depth = 0;
        int aim = 0;

        foreach (string s in data) {
            
            string[] dirX = s.Split(" ");

            switch (dirX[0]) {
                case "forward":
                    pos += int.Parse(dirX[1]);
                    depth += (aim * int.Parse(dirX[1]));
                    break;
                case "down":
                    aim += int.Parse(dirX[1]);
                    break;
                case "up":
                    aim -= int.Parse(dirX[1]);
                    break;
            }

        }

        Console.WriteLine(String.Format("Part Two | Position: {0} | Depth: {1} | Result: {2}", pos, depth, pos * depth));

    }

}

File: 2021/C/Day02/program.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#include "../buglLib/fileReader.h"
#include "../buglLib/stringManipulator.h"

int solve1(char** arr, int n) {

    int pos = 0;
    int depth = 0;

    for (int i = 0; i < n; i++) {

        char** command = split(arr[i], ' ');

        if (strcmp(command[0], "forward") == 0) {

            pos += atoi(command[1]);

        }
        else if (strcmp(command[0], "down") == 0) {

            depth += atoi(command[1]);

        }
        else if (strcmp(command[0], "up") == 0) {

            depth -= atoi(command[1]);

        }

        cleanMem(command, 2);

    }

    printf("Part One | Position: %d | Depth: %d\n", pos, depth);

    return (pos * depth);

}

int solve2(char** arr, int n) {

    int pos = 0;
    int depth = 0;
    int aim = 0;

    for (int i = 0; i < n; i++) {

        char** command = split(arr[i], ' ');

        if (strcmp(command[0], "forward") == 0) {

            pos += atoi(command[1]);
            depth += (aim * atoi(command[1]));

        }
        else if (strcmp(command[0], "down") == 0) {

            aim += atoi(command[1]);

        }
        else if (strcmp(command[0], "up") == 0) {

            aim -= atoi(command[1]);

        }

        cleanMem(command, 2);

    }

    printf("Part Two | Position: %d | Depth: %d\n", pos, depth);

    return (pos * depth);

}

int main(int argc, char** argv) {

    char** arr = readFile(argv[1]);
    int n = arrlen(arr);

    int res1 = solve1(arr, n);
    printf("Result: %d\n", res1);
    
    int res2 = solve2(arr, n);
    printf("Result: %d\n", res2);

    cleanMem(arr, MAX_LINES);

    return 0;

}